examples/lib/cache.js

```diff
@@ -1,7 +1,32 @@
 var setupCache = function() {
+  let newWorker;
+
+  function showUpdateModal() {
+    $('#update-prompt-modal').addClass('show');
+  }
+
+  $('#reload').on('click', function() {
+    newWorker.postMessage({ action: 'skipWaiting' });
+  });
+
   if ('serviceWorker' in navigator) {
     navigator.serviceWorker.register('sw.js', { scope: '/examples/' })
       .then(function(registration) {
+        registration.addEventListener('updatefound', () => {
```

Comment:
So, here we can detect if there's a new version, but we don't actually tie this to `getLatestVersionNumber()`, as far as I can read it? How might we do that - could we say `Click to install version X` here? Or is `getLatestVersionNumber()` not reading from the same source as `updatefound` here?

Reply:
We don't have to change things; i like how this looks. But let's leave enough inline comments to make the answer to this question really clear! Esp. because this is complicated stuff, so future coders should be able to follow what you've done. Thanks!